pkg/collector/collector.go

```diff
@@ -286,131 +230,60 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
-			v.PodName, v.Namespace,
-		)
-		ch <- desc_current
-
-		// de_total and desc_total give indexable values for total energy consumptions for all pods
-		desc_total := prometheus.MustNewConstMetric(
-			podDesc.AllAggr,
-			prometheus.CounterValue,
-			float64(v.AggEnergyInCore+v.AggEnergyInDram+v.AggEnergyInOther),
-			v.PodName, v.Namespace,
-		)
-		ch <- desc_total
-
-		// de_cpu_current and desc_cpu_current give indexable values for current CPU energy consumptions (in 3 seconds) for all pods
-		desc_cpu_current := prometheus.MustNewConstMetric(
-			podDesc.CPUCurr,
-			prometheus.GaugeValue,
-			float64(v.CurrEnergyInCore),
-			v.PodName, v.Namespace,
-		)
-		ch <- desc_cpu_current
-
-		// de_cpu_total and desc_cpu_total give indexable values for total CPU energy consumptions for all pods
-		desc_cpu_total := prometheus.MustNewConstMetric(
-			podDesc.CPUAggr,
-			prometheus.CounterValue,
-			float64(v.AggEnergyInCore),
-			v.PodName, v.Namespace,
-		)
-		ch <- desc_cpu_total
-
-		// de_dram_current and desc_dram_current give indexable values for current DRAM energy consumptions (in 3 seconds) for all pods
-		desc_dram_current := prometheus.MustNewConstMetric(
-			podDesc.DRAMCurr,
-			prometheus.GaugeValue,
-			float64(v.CurrEnergyInDram),
-			v.PodName, v.Namespace,
-		)
-		ch <- desc_dram_current
-
-		// de_dram_total and desc_dram_total give indexable values for total DRAM energy consumptions for all pods
-		desc_dram_total := prometheus.MustNewConstMetric(
-			podDesc.DRAMAggr,
-			prometheus.CounterValue,
-			float64(v.AggEnergyInDram),
-			v.PodName, v.Namespace,
-		)
-		ch <- desc_dram_total
-
-		// de_gpu_current and desc_gpu_current give indexable values for current GPU energy consumptions (in 3 seconds) for all pods
-		desc_gpu_current := prometheus.MustNewConstMetric(
-			podDesc.GPUCurr,
-			prometheus.GaugeValue,
-			float64(v.CurrEnergyInGPU),
-			v.PodName, v.Namespace,
-		)
-		ch <- desc_gpu_current
-
-		// de_gpu_total and desc_gpu_total give indexable values for total GPU energy consumptions for all pods
-		desc_gpu_total := prometheus.MustNewConstMetric(
-			podDesc.GPUAggr,
-			prometheus.CounterValue,
-			float64(v.AggEnergyInGPU),
-			v.PodName, v.Namespace,
-		)
-		ch <- desc_gpu_total
-
-		// de_other_current and desc_other_current give indexable values for current DRAM energy consumptions (in 3 seconds) for all pods
-		desc_other_current := prometheus.MustNewConstMetric(
-			podDesc.OtherCurr,
+		// ratio curr pod energy 
+		for ekey, elabel := range ENERGY_LABELS {
+			edesc := prometheus.MustNewConstMetric(
+				podDesc.CurrEnergy[elabel],
+				prometheus.GaugeValue,
+				v.GetPrometheusEnergyValue(ekey, true),
+				v.PodName, v.Namespace,
+			)
+			ch <- edesc
+		}
+		// dyn curr pod energy
+		edesc := prometheus.MustNewConstMetric(
+			podDesc.CurrEnergy[DYN_ENERGY_LABEL],
 			prometheus.GaugeValue,
-			float64(v.CurrEnergyInOther),
+			float64(v.DynEnergy.Curr)/1000.0, //J
```

Comment:
If switching to J from mJ, please also update the dashboard unit

Reply:
I'm sorry but could you clarify where do you mean the dashboard unit?

Reply:
the dashboard uses mW https://github.com/sustainable-computing-io/kepler/blob/main/grafana-dashboards/Kepler-Exporter-Dashboard-openshift.json#L133

Reply:
I see. Do you think should we export energy in J or mJ? 
If you agree to change to J, I will update the grafana. But, if you think it is better to keep the decimal, I will modify the collector sourcecode.

Reply:
let's keep mJ then, thanks

Reply:
Thanks. I see that originally you report node power as J. Should I keep it like that? 
So, the units will be
- node power -> J
- package power -> mJ
- pod power -> mJ 

do I understand it correctly?

Reply:
that sounds right